socket.go

```diff
@@ -30,12 +30,13 @@ var (
 
 // socket implements the ZeroMQ socket interface
 type socket struct {
-	ep    string // socket end-point
-	typ   SocketType
-	id    SocketIdentity
-	retry time.Duration
-	sec   Security
-	log   *log.Logger
+	ep       string // socket end-point
+	typ      SocketType
+	id       SocketIdentity
+	retry    time.Duration
+	sec      Security
+	log      *log.Logger
+	topicsCb func() []string
```

Comment:
```suggestion
	topics func() []string
```

also, could you add a test for this?

Reply:
`topics()` is already used for the pub topics, I renamed it to `subTopics`